### PR TITLE
External upload date

### LINKED ISSUE #101: User story for External Partners
- [x] Upload csv file for database

### PR DESCRIPTION
**Pull request by chk1994**
Fixes #101

### CODE REVIEW

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/model/Note.java [hunk 1]
```diff
@@ -44,4 +44,8 @@ public Note(User creator, User patient, String noteContent, boolean isVisibleToP
     public void setIsVisibleToPatient(boolean option){
         this.isVisibleToPatient = option;
     }
+
+    public void setNoteContent(String noteContent){
```
Comment (Zhiyuan-Amos) [inline backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/model/Note.java]: Put `@Setter` above `noteContent` will do.

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/payload/NoteUpdateRequest.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@Getter
+public class NoteUpdateRequest {
+
+    @NonNull
+    private Long noteID;
+
+    @NonNull
+    @NotBlank
+    private String noteContent;
```
Comment (Zhiyuan-Amos) [inline backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/payload/NoteUpdateRequest.java]: Should we limit the length of the noteContent?

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/controller/NoteController.java [hunk 3]
```diff
@@ -79,6 +80,26 @@
                 .body(new ApiResponse(true, "Note_" + note.getNoteID() + " permission changed"));
     }
 
+    @PostMapping("/checkNoteIdConsent/")
+    //Patient get notes permitted by any other therapists
+    public ResponseEntity<?> checkNoteIdConsent(@Valid @RequestBody NoteCheckPermissionRequest noteCheckPermission, @CurrentUser User currentUser) {
+
+        Note note = noteService.checkNoteIdConsent(noteCheckPermission, currentUser, 0, 30);
+
+        URI location = ServletUriComponentsBuilder
```
Comment (Zhiyuan-Amos) [inline backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/controller/NoteController.java]: do you know what's this is used for?

**Comment by Zhiyuan-Amos** [comment]
@chilipadiboy you should take a look at the first commit to see if it fits their requirements.

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/payload/NoteUpdateRequest.java (hunk 2 above)
Comment (chk1994) [reply backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/payload/NoteUpdateRequest.java]: oh ya I forgot to do it. Would 140 be sufficient for a short note? or isit too much?

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/model/Note.java (hunk 1 above)
Comment (chk1994) [reply backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/model/Note.java]: haha cause I didnt want to have unnecessary setters.

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/controller/NoteController.java (hunk 3 above)
Comment (chk1994) [reply backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/controller/NoteController.java]: the checkNoteIdConsent method?

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/model/Note.java (hunk 1 above)
Comment (Zhiyuan-Amos) [reply backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/model/Note.java]: If you put it above `noteContent`, then only `noteContent` will have the setter method; the other fields won't have their respective setter methods.

File: backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/controller/NoteController.java (hunk 3 above)
Comment (Zhiyuan-Amos) [reply backend-server/src/main/java/org/cs4239/team1/protectPMLeefrontendserver/controller/NoteController.java]: As in, is there a need to return the `location`?

**Comment by gjlam95** [comment]
@chk1994  API works, but wondering how do I find the uploaded .csv file?

**Comment by chk1994** [comment]
@gjlam95 What do you mean by find the uploaded file? The code api takes in a csv file, then read its content line by line in a while loop to add them into the DB by creating new users/records/notes. The csv file isn’t saved anywhere haha

**Comment by gjlam95** [comment]
@chk1994 icic. I'm able to get the response so i believe the file has been uploaded successfully

**Comment by chk1994** [comment]
haha yeah I have verified that the newly created users/records are inside after running successfully. However if any line in the csv got problem, it will throw an exception and stop there.

**Pull request merged**